PrototypeArmoury/Src/PrototypeArmoury/Classes/UIUtilities_PA.uc

```diff
@@ -153,21 +156,23 @@ static function OnStripUpgradesDialogCallback (Name eAction)
 			UnitState = XComGameState_Unit(UpdateState.ModifyStateObject(class'XComGameState_Unit', Soldiers[idx].ObjectID));
 			if (UnitState != none)
 			{
-				ItemState = UnitState.GetItemInSlot(eInvSlot_PrimaryWeapon);
-				WeaponTemplate = X2WeaponTemplate(ItemState.GetMyTemplate());
-				if (WeaponTemplate != none && WeaponTemplate.NumUpgradeSlots > 0)
+				foreach default.SlotsToStrip(SlotToStrip)
 				{
-					ItemState = XComGameState_Item(UpdateState.ModifyStateObject(class'XComGameState_Item', ItemState.ObjectID));
-					EquippedUpgrades = ItemState.GetMyWeaponUpgradeTemplates();
-
-					if (ShouldRemoveNicknamedUpgrades(ItemState))
+					ItemState = UnitState.GetItemInSlot(SlotToStrip);
```

Comment:
This assumes that all upgradable items are in single-item slots. For sake of completion, please rework this bit to support upgradable items in utility and other multi-item slots